src/app/file-panel.js

```diff
@@ -108,14 +112,16 @@ function filepanel (appAPI, files) {
 
   var events = new EventManager()
   var element = template()
-  // TODO please do not add custom javascript objects, which have no
-  // relation to the DOM to DOM nodes
-  element.events = events
+
   fileExplorer.events.register('focus', function (path) {
     appAPI.switchToFile(path)
   })
 
-  return element
+  var api = { events: events }
+
+  container.appendChild(element)
+
+  return api
```

Comment:
Here: https://github.com/ethereum/browser-solidity/pull/557/files#diff-bd9c9dcd314f2d7df52935b3a6a4d504R198 you use `filepanel` as the constructor of an object, but you return something explicitly in this line. Am I looking at two different things? If yes, they should not have the same name. If no, please just use something like `this.events = events` here and do not return explicitly.